lazy.md

```diff
@@ -13,3 +13,623 @@ author:
+   by stdexec.
+- `co_await just_error(error)` would be reasonable for completions
+    always indicating failure; supported by stdexec.
+- the result type is already inconsistent instead of always
+    `tuple<decay_t<T>...>` => support multiple `set_value_t`
+    completions and produce a `variant<tuple<T...>...>`
+- the last option is different from `co_await into_variant<sender>`
+    as it treats `set_stopped_t()` and `set_error_t(error)` differently
+- it may be reasonable to detect specific shapes of completion
+    signatures and turn them into something more coroutine friendly like
+    an `optional<T>` or an `expected<T, E>`; alternatively, support
+    corresponding algorithms like `into_optional`, `into_expected`
+
+Changes to the status quo would be applied to `as_awaitable(sender)`
+to get consistent behavior for user-define coroutine tasks using
+`as_awaitable` or `with_awaitable_sender`.
+
+To make the proposed `co_await` result type more concrete, here is an
+example using a custom sender type `custom` for which the completion signatures
+can be explicitly specified:
+
+    namespace ex = std::execution;
+
+    template <typename... Completions>
+    struct custom {
+        using sender_concept = ex::sender_t;
+        using completion_signatures = ex::completion_signatures<
+            ex::set_stopped_t(), Completions...
+        >;
+
+        template <ex::receiver R>
+        struct state {
+            using operation_state_concept = ex::operation_state_t;
+            std::decay_t<R> r;
+            void start() & noexcept { ex::set_stopped(std::move(r)); }
+        };
+
+        template <ex::receiver R>
+        state<R> connect(R&& r) { return { std::forward<R>(r) }; }
+    };
+
+    ex::lazy<void> fun() {
+                                                          // co_await result type
+        co_await ex::just();                              // void
+        int i = co_await ex::just(0);                     // int
+        auto[i, b, c] = co_await ex::just(0, true, 'c');  // tuple<int, bool, char>
+        auto v = co_await custom<ex::set_value_t(), ex::set_value_t(int)>;
+                                          // variant<tuple<>, tuple<int>>
+    }
+
+## Scheduler Affinity
+
+**TODO** turn into text
+
+**TODO** confirm whether it is possible have an inline scheduler which can
+    immediately start the operation (or if corresponding detection can go
+    into `continues_on`)
+
+- Resume the coroutine on the same scheduler even if the sender completes
+    on a different scheduler: `await_transform(sender)` returns
+    `continues_on(as_awaitable(sender), scheduler)`.
+- The used `scheduler` is determined using `get_scheduler(env)` where
+    `env` is the environment of the task's receiver => the `scheduler`
+    needs to be type-erased in some form
+- What should happen if there is no `get_scheduler(env)`? Require a
+    scheduler (unifex), use an inline scheduler (stdexec), or something
+    else?
```

Comment:
My ranked preferences are:
 1. require a scheduler, or
 2. something else.

I think "use an inline scheduler" would be a disaster.